# Event Count for Wind

In [2]:
import os
import pandas as pd
from pathlib import Path

w_combined_dir = Path('/Users/jimnguyen/IRMII/SCS_API/wind_reports/combined/')

w_yearly_totals = {}
w_monthly_counts = {}

for csv_file in w_combined_dir.glob('wind_*_combined.csv'):
    year = int(csv_file.stem.split('_')[1])
    df = pd.read_csv(csv_file)
    
    w_yearly_totals[year] = len(df)
    
    w_month_counts = df['Month'].value_counts().sort_index()
    w_full_month_counts = {m: w_month_counts.get(m, 0) for m in range(1, 13)}
    w_monthly_counts[year] = w_full_month_counts

#moth df
w_monthly_df = pd.DataFrame.from_dict(w_monthly_counts, orient='index')
w_monthly_df.index.name = 'Year'
w_monthly_df.columns = [f'Month_{m}' for m in w_monthly_df.columns]

w_monthly_df['Year_Total'] = w_monthly_df.sum(axis=1)
w_monthly_df.sort_index(inplace=True)

#year df
w_yearly_df = w_monthly_df['Year_Total']

In [3]:
w_monthly_df

Month_1  Month_2  Month_3  Month_4  Month_5  Month_6  Month_7  Month_8  \
Year                                                                           
2005      196       62      380      477      919     2888     2418     1564   
2006      138      111      846     1456     1719     1988     2745     1906   
2007      100      147      376      656     1138     2875     2004     2785   
2008      672      742      892      755     1849     4721     3403     1423   
2009       83      659      467      779     1445     3237     2175     1719   
2010      169       19      115      819     1601     4128     2885     2361   
2011       79      899      665     4545     3189     5632     4343     3478   
2012      469      598      747      747     1886     3573     5802     2244   
2013      908      135      547     1058     2459     4608     2844     1578   
2014      216      654      217     1137     1326     4862     3670     1506   
2015      101       60      129     1723     2049     4819     4189     2040   
2016      149      922      761      977     1758     3968     6385     1932   
2017      738      909     2241     1972     3487     3638     3653     1713   
2018       91      172      337     1229     3085     5008     3361     2596   
2019      120      214      638     2530     3320     5209     3888     4010   
2020     1101      707      718     3588     2234     4492     4794     3895   
2021       83       97      808      866     2315     3767     4133     3752   
2022      185      147     1366     1562     3921     5337     4864     2284   
2023      814      414     1574     1595     1404     5348     7689     4657   
2024      729      405      461     1897     5640     4650     5056     3792   

      Month_9  Month_10  Month_11  Month_12  Year_Total  
Year                                                     
2005      763        84       619        63       10433  
2006      489       280       234       175       12087  
2007      579       624        72       119       11475  
2008      187       106        75       327       15152  
2009      246       348         7       218       11383  
2010      702       782       353       158       14092  
2011     1044       215       263       130       24482  
2012     1684       263        86       516       18615  
2013     1037       661       634       416       16885  
2014      915       961       190       118       15772  
2015      536       159       365       465       16635  
2016      819       226       308       152       18357  
2017      636       514       350        50       19901  
2018      851       455       392       238       17815  
2019     1065      1132        60       220       22406  
2020      354       434       816       143       23276  
2021      859       597       112      1324       18713  
2022      574       243       333       165       20981  
2023     1594       224        36       121       25470  
2024      863       354       160       480       24487

# Event count for hail

In [4]:
import os
import pandas as pd
from pathlib import Path

h_combined_dir = Path('/Users/jimnguyen/IRMII/SCS_API/hail_reports/combined/')

h_yearly_totals = {}
h_monthly_counts = {}

for csv_file in h_combined_dir.glob('hail_*_combined.csv'):
    year = int(csv_file.stem.split('_')[1])
    df = pd.read_csv(csv_file)
    
    h_yearly_totals[year] = len(df)
    
    h_month_counts = df['Month'].value_counts().sort_index()
    h_full_month_counts = {m: h_month_counts.get(m, 0) for m in range(1, 13)}
    h_monthly_counts[year] = h_full_month_counts

#moth df
h_monthly_df = pd.DataFrame.from_dict(h_monthly_counts, orient='index')
h_monthly_df.index.name = 'Year'
h_monthly_df.columns = [f'Month_{m}' for m in h_monthly_df.columns]

h_monthly_df['Year_Total'] = h_monthly_df.sum(axis=1)
h_monthly_df.sort_index(inplace=True)

#year df
h_yearly_df = w_monthly_df['Year_Total']

In [5]:
h_monthly_df

Month_1  Month_2  Month_3  Month_4  Month_5  Month_6  Month_7  Month_8  \
Year                                                                           
2005       45      299     1065     1537     2079     2704     1119      852   
2006      215      161     1222     3296     2867     2268     1324      990   
2007        3      249      791     1302     1977     2517     1321     1324   
2008      267      360      857     1349     2682     4425     2023     1310   
2009       21      379      617     1607     1585     2898     1741      957   
2010       63       13      255      743     1493     1440      824      295   
2011       12      126      879     3089     3158     3142     1130     1351   
2012      111      232     1698     1570     2368     1444     1377      625   
2013       24      129      732      825     1889     1496     1036      925   
2014        7       56      213     1164     1817     2110     1052      565   
2015        0        2      471     1687     1148     1897      843      842   
2016       69      120      748     1250     2204     1107     1224      475   
2017      145      446     1252     1453     1916     1505      840      584   
2018       11       18      495      529     1927     1494     1026      700   
2019       16       37      388      647     2111     1589      990     1130   
2020       35       51      524     1354     1365     1215     1211      858   
2021       29       28      554      895     1236      876      703      639   
2022       18       34      201     1076     2050     1438      857      358   
2023       85       97      816     1847     1663     2785     2003      862   
2024       57      259     1032     1197     3190     1296      795      560   

      Month_9  Month_10  Month_11  Month_12  Year_Total  
Year                                                     
2005      729       139       303       219       11090  
2006      582       257        52        29       13263  
2007      383       313        76        35       10291  
2008      241       105       108        47       13774  
2009      306        73         3        22       10209  
2010      414       292        35        44        5911  
2011      316       145        76         4       13428  
2012      424       104        55        34       10042  
2013      322       124        50         1        7553  
2014      356       371         7        44        7762  
2015      374       124        66        65        7519  
2016      409       159        82        24        7871  
2017      200       158        80        11        8590  
2018      121        86        19        44        6470  
2019      596       114        33        30        7681  
2020      116        44        37         0        6810  
2021      287        89        51        99        5486  
2022      230       149        98        15        6524  
2023      624       197         8        96       11083  
2024      198        76        45        46        8751

# Event count for Tornado

In [11]:
import os
import pandas as pd
from pathlib import Path

t_combined_dir = Path('/Users/jimnguyen/IRMII/SCS_API/tornado_reports/combined/')

t_yearly_totals = {}
t_monthly_counts = {}

for csv_file in t_combined_dir.glob('torn_*_combined.csv'):
    year = int(csv_file.stem.split('_')[1])
    df = pd.read_csv(csv_file)
    
    t_yearly_totals[year] = len(df)
    
    t_month_counts = df['Month'].value_counts().sort_index()
    t_full_month_counts = {m: t_month_counts.get(m, 0) for m in range(1, 13)}
    t_monthly_counts[year] = t_full_month_counts

#moth df
t_monthly_df = pd.DataFrame.from_dict(t_monthly_counts, orient='index')
t_monthly_df.index.name = 'Year'
t_monthly_df.columns = [f'Month_{m}' for m in t_monthly_df.columns]

t_monthly_df['Year_Total'] = t_monthly_df.sum(axis=1)
t_monthly_df.sort_index(inplace=True)

#year df
t_yearly_df = w_monthly_df['Year_Total']

In [12]:
t_monthly_df

Month_1  Month_2  Month_3  Month_4  Month_5  Month_6  Month_7  Month_8  \
Year                                                                           
2005       32        9       73      151      142      303      123      104   
2006       44       15      219      319      164      130       68       95   
2007       29       87      211      181      282      149       50       78   
2008      136      230      150      189      597      394      120      145   
2009       10       44      123      270      227      299      134       63   
2010       41        1       37      197      290      428      173       70   
2011       10       67      115     1091      416      196      100       55   
2012      110       76      271      271      161      128       30       59   
2013       94       54       24       97      288      172       63       42   
2014        4       45       30      270      173      372       99       33   
2015       32        2       15      204      459      219      133       48   
2016       22      184      125      133      292      104      111      133   
2017      171      133      192      259      329      191       93      167   
2018       16       49       63      166      183      182      111       94   
2019       19       34      169      352      650      228      114       88   
2020       97       71      114      406      151      121      129      202   
2021       19       11      207       78      335      117      178      150   
2022       57       19      372      304      234      138       89       39   
2023      204       67      302      140      221      287      128      166   
2024       52       63       70      455      648      200      245      116   

      Month_9  Month_10  Month_11  Month_12  Year_Total  
Year                                                     
2005      112        23       126        18        1216  
2006       91        63        49        39        1296  
2007       62       115         7        25        1276  
2008      133        28        20        52        2194  
2009       10        69         4        57        1310  
2010       60       121        63        76        1557  
2011       76        30        66        18        2240  
2012       57        44         9        93        1309  
2013       17        83       141        20        1095  
2014       48        76        30        36        1216  
2015       29        67       109       117        1434  
2016       40        27        85        14        1270  
2017       36        99        53        19        1742  
2018      127       176        66        75        1308  
2019      102        73        21        91        1941  
2020       45        22        34        39        1431  
2021       33       163        28       237        1556  
2022       20        49       141       117        1579  
2023       23        36        22        60        1656  
2024       93        62        44       167        2215

# Event count using annual data set

## By month

In [19]:
import os
import glob
import pandas as pd


In [20]:
filtered_dir = '/Users/jimnguyen/IRMII/SCS_API/data/Storm_Reports/filtered'

paths = glob.glob(os.path.join(filtereded_dir := filtered_dir, 'Storm_Reports_*_latlong.csv'))

dfs = []
for p in paths:
    df_year = pd.read_csv(p)
    dfs.append(df_year)

all_events = pd.concat(dfs, ignore_index=True)

all_events['MONTH_NUM'] = pd.to_datetime(all_events['MONTH_NAME'], format='%B').dt.month

grouped = (
    all_events
    .groupby(['YEAR', 'MONTH_NUM', 'EVENT_TYPE'])
    .size()
    .reset_index(name='Count')
)

pivot = (
    grouped
    .pivot_table(
        index=['YEAR', 'MONTH_NUM'],
        columns='EVENT_TYPE',
        values='Count',
        fill_value=0
    )
)

pivot = pivot.sort_index(level=[0, 1])
pivot


EVENT_TYPE       Hail  High Wind  Strong Wind  Thunderstorm Wind  Tornado
YEAR MONTH_NUM                                                           
1950 1            0.0        0.0          0.0                0.0      8.0
     2            0.0        0.0          0.0                0.0     27.0
     3            0.0        0.0          0.0                0.0     24.0
     4            0.0        0.0          0.0                0.0     17.0
     5            0.0        0.0          0.0                0.0     64.0
...               ...        ...          ...                ...      ...
2024 8          649.0       43.0         63.0             3179.0     98.0
     9          257.0      273.0         95.0              664.0     75.0
     10          98.0      264.0          4.0              268.0     66.0
     11          51.0      281.0         82.0              178.0     74.0
     12          58.0      518.0        250.0              294.0    147.0

[896 rows x 5 columns]

In [60]:
filtered_dir = '/Users/jimnguyen/IRMII/SCS_API/data/Storm_Reports/filtered'

paths = glob.glob(os.path.join(filtered_dir, 'Storm_Reports_*_latlong.csv'))

summary_list = []

for path in paths:
    basename = os.path.basename(path)
    parts = basename.split('_')
    year = int(parts[2])
    
    df = pd.read_csv(path)

    counts = df['EVENT_TYPE'].value_counts().to_dict()
    
    row = {'Year': year}
    for ev_type, cnt in counts.items():
        row[ev_type] = cnt
    summary_list.append(row)

summary_df = pd.DataFrame(summary_list).fillna(0).sort_values('Year').reset_index(drop=True)

for col in summary_df.columns:
    if col != 'Year':
        summary_df[col] = summary_df[col].astype(int)

summary_df = summary_df.set_index('Year').sort_index()

summary_df

Hail  Thunderstorm Wind  High Wind  Tornado  Strong Wind
Year                                                           
1950      0                  0          0      223            0
1951      0                  0          0      269            0
1952      0                  0          0      272            0
1953      0                  0          0      492            0
1954      0                  0          0      609            0
...     ...                ...        ...      ...          ...
2020   7677              19315       3682     1251         1193
2021   6272              15533       4599     1545          863
2022   7180              17595       5179     1383         1066
2023  11767              21248       3491     1523         1101
2024   8992              20064       3993     2141         1314

[75 rows x 5 columns]

In [126]:
summary_df['Wind'] = summary_df['Thunderstorm Wind'] + summary_df['High Wind'] + summary_df['Strong Wind']
summary_df

Hail  Thunderstorm Wind  High Wind  Tornado  Strong Wind   Wind
Year                                                                  
1950      0                  0          0      223            0      0
1951      0                  0          0      269            0      0
1952      0                  0          0      272            0      0
1953      0                  0          0      492            0      0
1954      0                  0          0      609            0      0
...     ...                ...        ...      ...          ...    ...
2020   7677              19315       3682     1251         1193  24190
2021   6272              15533       4599     1545          863  20995
2022   7180              17595       5179     1383         1066  23840
2023  11767              21248       3491     1523         1101  25840
2024   8992              20064       3993     2141         1314  25371

[75 rows x 6 columns]

# Now we find the error 

We will find both the absolute error and the percentage error

### Error By Year

### Wind

In [127]:
wind_annual = summary_df.loc[2005:2019, 'Wind']
wind_monthly_sum = overlap_w_monthly_df.loc[2005:2019, 'Year_Total']

wind_error_df = pd.DataFrame({
    'Wind_Annual': wind_annual,
    'Wind_MonthlySum': wind_monthly_sum
})
wind_error_df['Absolute Error'] = wind_error_df['Wind_Annual'] - wind_error_df['Wind_MonthlySum']
wind_error_df['Percentage Error'] = wind_error_df['Absolute Error']/wind_error_df['Wind_Annual']

wind_error_df = wind_error_df.reset_index().rename(columns={'index': 'Year'})

wind_error_df


Year  Wind_Annual  Wind_MonthlySum  Absolute Error  Percentage Error
0   2005        15233            10433            4800          0.315105
1   2006        18093            12087            6006          0.331952
2   2007        16010            11475            4535          0.283260
3   2008        22240            15152            7088          0.318705
4   2009        17675            11383            6292          0.355983
5   2010        19548            14092            5456          0.279108
6   2011        25585            24482            1103          0.043111
7   2012        21071            18615            2456          0.116558
8   2013        17435            16885             550          0.031546
9   2014        17914            15772            2142          0.119571
10  2015        17304            16635             669          0.038662
11  2016        19681            18357            1324          0.067273
12  2017        21104            19901            1203          0.057003
13  2018        18622            17815             807          0.043336
14  2019        23997            22406            1591          0.066300

In [128]:
wind_average_error = wind_error_df['Percentage Error'].sum()/15
wind_average_error

np.float64(0.16449824179615455)

## Error for hail

In [146]:
hail_annual = summary_df.loc[2005:2019, 'Hail']
hail_monthly_sum = h_monthly_df.loc[2005:2019, 'Year_Total']

hail_error_df = pd.DataFrame({
    'Hail_Annual': hail_annual,
    'Hail_MonthlySum': hail_monthly_sum
})

hail_error_df['Absolute Error'] = hail_error_df['Hail_Annual'] - hail_error_df['Hail_MonthlySum']
hail_error_df['Percentage Error'] = hail_error_df['Absolute Error']/ hail_error_df['Hail_Annual']

hail_error_df = hail_error_df.reset_index().rename(columns={'index': 'Year'})

hail_error_df


Year  Hail_Annual  Hail_MonthlySum  Absolute Error  Percentage Error
0   2005        13808            11090            2718          0.196842
1   2006        16656            13263            3393          0.203710
2   2007        12712            10291            2421          0.190450
3   2008        17768            13774            3994          0.224786
4   2009        13313            10209            3104          0.233156
5   2010        10918             5911            5007          0.458600
6   2011        17331            13428            3903          0.225203
7   2012        13385            10042            3343          0.249757
8   2013         9625             7553            2072          0.215273
9   2014         9964             7762            2202          0.220996
10  2015         9398             7519            1879          0.199936
11  2016         9407             7871            1536          0.163283
12  2017        10398             8590            1808          0.173880
13  2018         7863             6470            1393          0.177159
14  2019         8992             7681            1311          0.145796

In [147]:
hail_average_error = hail_error_df['Percentage Error'].sum()/15
hail_average_error

np.float64(0.21858848923694257)

## Error for tornado


In [144]:
tor_annual = summary_df.loc[2005:2019, 'Tornado']
tor_monthly_sum = t_monthly_df.loc[2005:2019, 'Year_Total']

tor_error_df = pd.DataFrame({
    'Tor_Annual': tor_annual,
    'Tor_MonthlySum': tor_monthly_sum
})

tor_error_df['Absolute Error'] = tor_error_df['Tor_Annual'] - tor_error_df['Tor_MonthlySum']
tor_error_df['Percentage Error'] = tor_error_df['Absolute Error']/ tor_error_df['Tor_Annual']

tor_error_df = tor_error_df.reset_index().rename(columns={'index': 'Year'})

tor_error_df


Year  Tor_Annual  Tor_MonthlySum  Absolute Error  Percentage Error
0   2005        1343            1216             127          0.094564
1   2006        1263            1296             -33         -0.026128
2   2007        1242            1276             -34         -0.027375
3   2008        1954            2194            -240         -0.122825
4   2009        1273            1310             -37         -0.029065
5   2010        1449            1557            -108         -0.074534
6   2011        2074            2240            -166         -0.080039
7   2012        1058            1309            -251         -0.237240
8   2013        1053            1095             -42         -0.039886
9   2014        1064            1216            -152         -0.142857
10  2015        1320            1434            -114         -0.086364
11  2016        1081            1270            -189         -0.174838
12  2017        1651            1742             -91         -0.055118
13  2018        1254            1308             -54         -0.043062
14  2019        1734            1941            -207         -0.119377

In [145]:
tor_average_error = tor_error_df['Percentage Error'].sum()/15
tor_average_error

np.float64(-0.07760962998032696)

## Summary

Average error for each year rounded to 3 digits percentage

Wind: 16.45%
Hail: 21.859%
Tornado: 7.761%